src/reactionsystem.jl

```diff
@@ -516,6 +516,10 @@ struct ReactionSystem{V <: NetworkProperties} <:
     function ReactionSystem(eqs, rxs, iv, sivs, states, spcs, ps, var_to_name, observed,
                             name, systems, defaults, connection_type, nps, cls, cevs, devs,
                             complete::Bool = false; checks::Bool = true)
+        # Filters away any potential obervables from `states` and `spcs`.
+        obs_vars = [obs_eq.lhs for obs_eq in observed]
+        states = filter(state -> !any(isequal(state, obs_var) for obs_var in obs_vars), states)
+        spcs = filter(spc -> !any(isequal(spc, obs_var) for obs_var in obs_vars), spcs)
```

Comment:
I don't think this is the place to do this. It would be reasonable to have a check and error message here, but this shouldn't rearrange a user's state and species vectors.

Can you move this into the DSL logic, and make this just a consistency check that `states` and `spcs` don't contain an observable?

Reply:
This has been moved now